neo/Ledger/MemoryPool.cs

```diff
@@ -337,18 +342,30 @@ internal void InvalidateVerifiedTransactions()
         // Note: this must only be called from a single thread (the Blockchain actor)
         internal void UpdatePoolForBlockPersisted(Block block, Snapshot snapshot)
         {
+            bool policyChanged = LoadPolicy(snapshot);
+
             _txRwLock.EnterWriteLock();
             try
             {
-                // First remove the transactions verified in the block.
-                foreach (Transaction tx in block.Transactions)
+                if (policyChanged)
                 {
-                    if (TryRemoveVerified(tx.Hash, out _)) continue;
-                    TryRemoveUnVerified(tx.Hash, out _);
+                    _unsortedTransactions.Clear();
```

Comment:
It is very strange to clean and lose them, why not cache just cache and keep the ones that are still valid?

Reply:
I mean, isn't anyone paying any extra free that will still be kept valid?

It looks strange to flush mempool.
Furthermore, because I imagine a game theory scenario, very dynamic.

Reply:
If you imagine that everyone is paying the exact fee, then all txs will be abandoned brother. Even if someone paid an extra fee, the only way to find out without recomputing everything is using Max-Trees... when you could find it in O(ln N) time, however pre-compute cost is so high, so as complication to do that, that's simply not worth it. Better to drop the whole pool.

Reply:
I do not know, it will not be more costly than force valid txs to be rebrocasted.

Reply:
Ok, you want a simpler idea, to scratch your mind, here is goes xD
Tx 1: size = 100  fee= 1000 GAS  (paid feePerByte = 10 GAS/byte)
Tx 2: size = 10  fee= 200 GAS  (paid feePerByte = 20 GAS/byte)
Tx 3: size = 1000  fee= 25000 GAS  (paid feePerByte = 25 GAS/byte)

Imagine that policy is now 10 GAS/byte, so all three are valid. If you increase policy to 15 GAS/byte, then you already know that Tx 1 will be dropped, but the others (Tx2 and Tx3) won't. You just need O(N) to scan the whole mempool, but you're right, you won't need to drop all of them, and rebroadcast ;) smart brother

Reply:
Maybe we can re-tell these transactions to `Blockchain`?

Reply:
That would be nice, Erik.

Maybe delete the ones with `feeperbyte` lower than the new policy and then normally call `reverify`. I think it will be almost the same, right?

Reply:
But no worries, Erik. If you want to merge we try that in another PR later.

Reply:
Done.